src/implementations/twig/components/site-header/site-header-language-switcher.html.twig

```diff
@@ -82,18 +82,15 @@
       >
         {{- _language_selector.overlay.title -}}
       </div>
-      {% include '@ecl/button/button.html.twig' with {
-        label: _language_selector.overlay.close_label,
-        variant: 'ghost',
-        icon: {
-          path: _icon_file_path,
-          name: 'close-filled',
-          size: 's'
-        },
-        hide_label: true,
-        extra_classes: 'ecl-site-header__language-close',
-        extra_attributes: [{ name: 'data-ecl-language-list-close' }],
-      } only %}
+      {% if _language_selector.overlay.close is not empty %}
+        {% include '@ecl/button/button.html.twig' with _language_selector.overlay.close|merge({
+          variant: 'ghost',
+          extra_classes: 'ecl-site-header__language-close',
```

Comment:
i think it's ok like this, i mean that should we expect extra classes or extra attributes for this button? It would be better, in theory, to always merge the data coming from the user with our internal definition of an include.
Here you merge, but then you set extra_classes and extra_attributes, if this are present in the user data they would be overridden.

Reply:
I agree that in theory it should be the case for each include we do, in every component. But that would make the twig template way bigger.
So if we have an easy way to do so, without having to write 4 or 5 lines each times, we could do it

Reply:
it's twig, might not look beatiful but it's not even that we have to do something that complex to achieve this, in this case it would be like:
```
        {% include '@ecl/button/button.html.twig' with _language_selector.overlay.close|merge({
          variant: 'ghost',
          extra_classes: _language_selector.overlay.close.extra_classes|default('') ~ ' ecl-site-header__language-close',
          extra_attributes: _language_selector.overlay.close.extra_attributes|default([])|merge([{
            name: 'data-ecl-language-list-close',
          }]),
        }) only %}
```
another approach (we have already many situations handled like this) would be to process the parameters we want to alter separately to combine the existing values with the ones we want to add and then use them in the include, but this is a bit more verbose.